variables.tf

```diff
@@ -49,51 +44,60 @@ variable "dns_servers" {
   description = "The DNS servers to be used with vNet."
 }
 
-variable "nsg_ids" {
-  type = map(string)
-  default = {
-  }
-  description = "A map of subnet name to Network Security Group IDs"
-}
-
-variable "route_tables_ids" {
-  type        = map(string)
-  default     = {}
-  description = "A map of subnet name to Route table ids"
-}
+variable "subnets" {
+  description = "A mapping of subnet names to their configurations."
+  type = map(object({
+    prefix = string
+    delegation = optional(object({
+      name    = string
+      actions = list(string)
+    }), null)
+    service_endpoints                             = optional(list(string), []),
+    private_endpoint_network_policies_enabled     = optional(bool, false)
+    private_link_service_network_policies_enabled = optional(bool, false)
+    network_security_group_id                     = optional(string, null)
+    route_table_id                                = optional(string, null)
+    route_table_alias                             = optional(string, null)
+  }))
+  default = {}
 
-variable "subnet_delegation" {
-  type        = map(map(any))
-  default     = {}
-  description = "A map of subnet name to delegation block on the subnet"
+  validation {
+    condition     = alltrue([for subnet_name, subnet in var.subnets : subnet.route_table_id == null || subnet.route_table_alias == null ? true : subnet.route_table_id == null && subnet.route_table_alias == null ? true : false])
```

Comment:
`subnet.route_table_id == null || subnet.route_table_alias == null`

if I'm reading the conditional right this will be true when both are defined, meaning the validation would succeed, contrary to the description

Reply:
unfortunately terraform does not have a native XOR function, in the past I've defined both cases explicitly when they are mutually exclusive
i.e. `(subnet.route_table_id != null && subnet.route_table_alias == null) || ( subnet.route_table_id == null && subnet.route_table_alias != null)`

Reply:
but since we allow them to both be null this could be simplified as `!(subnet.route_table_id != null && subnet.route_table_alias != null)`

Reply:
Updated in `47ab890`.